frontend/src/app/homepage-offer/offers.service.ts

```diff
@@ -1,38 +1,51 @@
 import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
-
 import 'rxjs/add/operator/map';
 import { Observable } from 'rxjs/Observable';
 
 import { environment } from '../../environments/environment';
-import { Offer } from './offers.model';
+import { ApiOffer, AppOffer } from './offers.model';
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
 
 
 @Injectable()
 export class OffersService {
   private url = `${environment.apiRoot}/offers/`;
+  private _offer$ = new BehaviorSubject<ApiOffer>(null);
+  public offer$ = this._offer$.asObservable();
 
   constructor (private http: HttpClient) { }
 
-  getOffers(): Observable<Offer[]> {
-    return this.http.get<Offer[]>(this.url)
-      .map(offers => offers.map(offer => this.loadDefaultImage(offer)));
+  getOffers(): Observable<ApiOffer[]> {
+    return this.http.get<ApiOffer[]>(this.url)
+      .map(offers => {
+        return offers.map(offer => this.loadDefaultImage(offer))
+      });
   }
 
-  getOffer(id: number): Observable<Offer> {
-    return this.http.get<Offer>(`${this.url}${id}/`)
+  getOffer(id: number): Observable<ApiOffer> {
+    return this.http.get<ApiOffer>(`${this.url}${id}/`)
       .map(offer => this.loadDefaultImage(offer));
   }
 
-  loadDefaultImage(offer: Offer): Offer {
+  loadDefaultImage(offer: ApiOffer): ApiOffer {
     if (!offer.image) {
         offer.image = 'assets/img/banner/volontulo_baner.png';
+        this._offer$.next(offer);
```

Comment:
Please delete this line as well. 

If we really needed to send the next event to offers, we shouldn't do it here, as this is just a helper method for images. We could do it instead in getOffer method after loadDefault image mapping. However, we won't send any events as we don't need those subjects here.

Reply:
If the offer is a subject, then it's actually a very good idea to use `next` here. However, only without mutating offer with setting `offer.image`, but with immutable new object (sth like `next({...offer, image: 'sth'})`). This way you can use `OnPush`, which is a real benefit. Mutating objects is usually hard to debug and has a performance penalty related to change detection.

Reply:
As I wrote in a previous comment, this is a utilily/helper method and it should not send next() events. Actually, this method has been moved to offer.utilities file in a recently merged PR from Julia, so after @weronikafelczak  merges from master it will disappear entirely from this file. (Weronika: possible merge conflict here)

@radoslawroszkowiak As of immutability, I do agree that it gives some benefits in a long run, however mutating state of not referenced objects is fine, especially when they arrive from backend and need some transformation before pushing them to components. Immutability makes sense for objects that are referenced by another objects, e.g when they get assigned to component. At this point that we are discussing here the mutated object is not referenced yet, so it would be garbage collected anyway. First time that they are referenced is when the template binds them to component after async pipe resolve, which is after aforementioned mutation.